src/hotspot/share/gc/shared/memAllocator.cpp

```diff
@@ -84,7 +101,9 @@ class MemAllocator::Allocation: StackObj {
   ~Allocation() {
     if (!check_out_of_memory()) {
       verify_after();
-      notify_allocation(_thread);
+      if (should_notify_allocation()) {
+        notify_allocation(_thread);
+      }
```

Comment:
`notify_allocation` used to call   `notify_allocation_low_memory_detector` and `notify_allocation_dtrace_sampler` without performing any "should notify" filtering. Is this change in behavior intentional?

Reply:
Not really, but I wonder if we actually need to do these for inside-TLAB allocs. I think I'll go the way Albert leans, and see what happens here.